### PR TITLE
Major CNFuzz refactor

### LINKED ISSUE #8: Remove cmdline flags and use ConfigMaps
Currently `cnfuzz` heavily relies on its configuration being passed as command line arguments.

Since `cnfuzz` is specifically meant for being deployed in Kubernetes switching to Kubernetes `ConfigMaps` will reduce Helm `Chart` complexity or any other Kubernetes deployment management tooling.

### LINKED ISSUE #63: Use flag package for command line argument parsing in Golang
Switch from Viper/Cobra to `flag` package of Golang and also support subcommands like setting a `mode` for CNFuzz.

### PR DESCRIPTION
**Pull request by Luukvdm**
Major refactor of CNFuzz. The biggest difference is in how config works. Instead of using flags we now simply parse a yaml file inside a config map. Other big change is in how logging is implemented. We now use logr in combination with zap (just like the Kubernetes project).

Some small things still need to be done. Like adding back the health checks and adding tests for the new code.

closes #8 
closes #63 (not entirely true, but I'm fine with keeping Cobra)
Probably fixes some bugs

### CODE REVIEW

**Comment by Luukvdm** [comment]
apparently the pipeline also needs to be updated

File: chart/cnfuzz/values.yaml [hunk 1]
```diff
@@ -44,23 +44,33 @@ imagePullSecrets: []
 nameOverride: ""
```
Comment (pcktdmp) [inline chart/cnfuzz/values.yaml:44]: Did you verify that moving the directory does not break the expected chart uri?

**Comment by pcktdmp** [comment]
Except the verification of helm chart path; LGTM!

File: chart/cnfuzz/values.yaml (hunk 1 above)
Comment (Luukvdm) [reply chart/cnfuzz/values.yaml:44]: I think I checked most paths, what uri are you thinking of?

**Pull request merged**